src/orca-jedi/geometry/Geometry.cc

```diff
@@ -27,16 +27,18 @@ oops::Variables orcaVariableFactory(const eckit::Configuration & config) {
   OrcaGeometryParameters params;
   params.validateAndDeserialize(config);
 
+  oops::Variables variables{};
   std::vector<std::string> names{};
   for (const NemoFieldParameters& nemoVariable :
         params.nemoFields.value()) {
-    std::string  name = nemoVariable.name.value();
+    std::string name = nemoVariable.name.value();
     if (std::find(names.begin(), names.end(), name) == names.end()) {
-      names.push_back(name);
+      names.emplace_back(name);
```

Comment:
Probably can get rid of `names` altogether here, `Variables::push_back` should now check if the variable already exists in `Variables` and only add new ones.